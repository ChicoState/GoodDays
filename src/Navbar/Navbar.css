.navbar {
    display: flex; /* Align items horizontally */
    justify-content: space-between; /* Space between brand and links */
    align-items: center; /* Center vertically */
    background-color: #333; /* Navbar background color */
    padding: 1rem; /* Padding around the navbar */
}

.navbar-brand h1 {
    color: #fff; /* Brand text color */
    margin: 0; /* Remove default margin */
}

.navbar-links {
    list-style-type: none; /* Remove bullet points from list */
    display: flex; /* Align list items horizontally */
    margin: 0; /* Remove default margin */
    padding: 0; /* Remove default padding */
}

.navbar-links li {
    margin: 0 1rem; /* Space between links */  
}

.navbar-links a {
    color: #fff; /* Link text color */
    text-decoration: none; /* Remove underline */
    transition: color 0.3s; /* Smooth color transition */
}

.navbar-links a:hover {
    text-decoration: underline; /* Underline on hover */
    color: #ffa500; /* Change color on hover */
}

/* Responsive styles for mobile */
@media (max-width: 768px) {
    .navbar-links {
        display: none; /* Hide links on mobile */
    }

    .navbar.active .navbar-links {
        display: flex; /* Show links when navbar is active */
        flex-direction: column; /* Stack links vertically */
        position: absolute;
        background-color: #333; /* Match navbar background */
        top: 60px; /* Adjust based on navbar height */
        right: 0;
        width: 100%;
    }

    .navbar-toggle {
        display: block; /* Show toggle button on mobile */
        cursor: pointer; /* Pointer on hover */
        color: #fff; /* Toggle button color */
    }
}