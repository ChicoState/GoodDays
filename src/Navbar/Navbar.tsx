import './Navbar.css';
import { Link } from "react-router-dom";

type NavbarProps = {
  //
};

const Navbar = (props: NavbarProps) => {
    return (
    <nav className={"navbar"}>
      <div className="navbar-brand">
        <h1>GoodDays</h1>
      </div>


      <ul className="navbar-links">
        <li><Link to="/">Home</Link></li>
        <li><Link to="/Create">Create</Link></li>
        <li><Link to="/Reports">Reports</Link></li>
      </ul>
    </nav>);
}

export default Navbar;
