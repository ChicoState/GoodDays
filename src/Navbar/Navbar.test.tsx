import * as React from "react";
import {render, screen} from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Navbar from "./Navbar";


describe("Navbar", () => {
    test("matches snapshot", () => {
        const {asFragment} = render(<BrowserRouter><Navbar></Navbar></BrowserRouter>);
        expect(asFragment()).toMatchSnapshot();
    });
});
