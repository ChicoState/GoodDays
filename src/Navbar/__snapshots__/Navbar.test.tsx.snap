// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
  >
    <div
      class="navbar-brand"
    >
      <h1>
        GoodDays
      </h1>
    </div>
    <ul
      class="navbar-links"
    >
      <li>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/Create"
        >
          Create
        </a>
      </li>
      <li>
        <a
          href="/Reports"
        >
          Reports
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
